experimental/dds/tree2/src/test/shared-tree/editing.spec.ts

```diff
@@ -195,143 +200,108 @@ describe("Editing", () => {
-			const x = insert(tree3, 0, "x");
-
-			const sequenced = sequencer.sequence([z, t, c, s, b, y, a, x, r]);
-			tree1.receive(sequenced);
-			tree2.receive(sequenced);
-			tree3.receive(sequenced);
-			tree4.receive(sequenced);
-
-			expectJsonTree(
-				[tree1, tree2, tree3, tree4],
-				["a", "b", "c", "r", "s", "t", "x", "y", "z"],
-			);
-		});
-
-		// TODO: Enable once local branch repair data is supported
-		it.skip("revert-only revive", () => {
-			const sequencer = new Sequencer();
-			const tree1 = TestTree.fromJson(["a", "b", "c"]);
-			const tree2 = tree1.fork();
-
-			const delB = remove(tree1, 1, 1);
-
-			const delABC = remove(tree2, 0, 3);
-
-			const seqDelB = sequencer.sequence(delB);
-			const seqDelABC = sequencer.sequence(delABC);
-
-			const revABC = tree2.runTransaction((forest, editor) => {
-				const field = editor.sequenceField({
-					parent: undefined,
-					field: rootFieldKeySymbol,
-				});
-				field.revive(0, 3, seqDelABC.revision, fakeRepair, 1);
-			});
-
-			const seqRevABC = sequencer.sequence(revABC);
-			const sequenced = [seqDelB, seqDelABC, seqRevABC];
-			tree1.receive(sequenced);
-			tree2.receive(sequenced);
-
-			expectJsonTree([tree1, tree2], ["a", "c"]);
+			const tree = makeTreeFromJson([]);
+			const c = tree.fork();
+			const t = tree.fork();
+			const z = tree.fork();
+
+			insert(c, 0, "c");
+			const b = c.fork();
+			insert(b, 0, "b");
+			const a = b.fork();
+			insert(a, 0, "a");
+
+			insert(t, 0, "t");
+			const s = t.fork();
+			insert(s, 0, "s");
+			const r = s.fork();
+			insert(r, 0, "r");
+
+			insert(z, 0, "z");
+			const y = z.fork();
+			insert(y, 0, "y");
+			const x = y.fork();
+			insert(x, 0, "x");
+
+			tree.merge(z);
+			tree.merge(t);
+			tree.merge(c);
+			tree.merge(s);
+			tree.merge(b);
+			tree.merge(y);
+			tree.merge(a);
+			tree.merge(x);
+			tree.merge(r);
+
+			a.rebaseOnto(tree);
+			r.rebaseOnto(tree);
+			x.rebaseOnto(tree);
+
+			expectJsonTree([tree, a, r, x], ["a", "b", "c", "r", "s", "t", "x", "y", "z"]);
 		});
 
 		// TODO: Enable once local branch repair data is supported
 		it.skip("intentional revive", () => {
-			const sequencer = new Sequencer();
-			const tree1 = TestTree.fromJson(["a", "b", "c"]);
+			const tree1 = makeTreeFromJson(["A", "B", "C"]);
 			const tree2 = tree1.fork();
 
-			const delB = remove(tree1, 1, 1);
-
-			const delABC = remove(tree2, 0, 3);
+			remove(tree1, 1, 1);
 
-			const seqDelB = sequencer.sequence(delB);
-			const seqDelABC = sequencer.sequence(delABC);
+			remove(tree2, 0, 3);
+			tree2.undo();
 
-			const revABC = tree2.runTransaction((forest, editor) => {
-				const field = editor.sequenceField({
-					parent: undefined,
-					field: rootFieldKeySymbol,
-				});
-				field.revive(0, 3, seqDelABC.revision, fakeRepair, 1, true);
-			});
-
-			const seqRevABC = sequencer.sequence(revABC);
-			const sequenced = [seqDelB, seqDelABC, seqRevABC];
-			tree1.receive(sequenced);
-			tree2.receive(sequenced);
```

Comment:
This section seems missing from the adjusted version: is it important?

Reply:
The transaction part has been replaced by a call to `undo`.

The rest (lines 178-281) are replaces by the `merge` and `rebaseOnto` calls.